### ISSUE TITLE
Is it possible to have a linux distro with this as the main interface?

### ISSUE THREAD

**Issue by technomancer21** [GitSquared/edex-ui#908]
I've been wanting to tinker around with the commandline side of linux, and learn to be a power user.
Frankly, while it might not be the most practical idea, I think setting up a distro that uses edex exclusively as a halfway point between a pure CLI and a desktop would be fun.
Is it possible to get a dependency list, so I could try this?

**Comment by GitSquared** [comment]
Somebody tried a few years back, I wrote this gist for them: https://gist.github.com/GitSquared/036678f0281a38ddd13e5df70cd0276b

This was for a blank Debian install, you may have to adjust it for whatever distro you choose.

**Comment by technomancer21** [comment]
Many thanks! gonna try with ubuntu mini.

**Comment by technomancer21** [comment]
So I ran into a small issue in running the appimage. when it starts the X server and runs the appimage, it just asks if I want to integrate edex into the system, it doesn't actually launch edex.

**Comment by GitSquared** [comment]
Isnt there a --flag in appimage to bypass this prompt?

**Comment by technomancer21** [comment]
I don't see anything resembling such I'm afraid. nor would i honestly know where to look. I don't have much experience with appimage containers.

**Comment by GitSquared** [comment]
Maybe here: https://discourse.appimage.org/t/stop-appimage-from-asking-to-integrate/488

**Comment by technomancer21** [comment]
That solved the integration issue, but now it just gives me a blank screen instead of launching. starting to wonder if i'm still missing a dependency. it asked me before to install zenity / kdialog / xdialog. i chose zenity. should i have installed all three? because xdialog wasnt found in the ubuntu repos.

**Comment by lebarde** [comment]
Maybe it lacks dependancies with `g++-multilib`. If such, you might also like to run `sysctl kernel.unprivileged_userns_clone=1` as root.

**Comment by technomancer21** [comment]
A good suggestion but unfortunately missed the mark. installing multilib didn't get me anywhere, neither did the other command. 
I am using the 2.2.2 appimage, however I left it for a good hour to see if it would eventually start and I still had no luck.
If it helps any, I am running this in a virtualbox VM for testing, easier to wipe and start fresh if I screw up than on a bare metal box.

**Comment by JohnDanielScanlon** [comment]
I misread this and thought you were requesting one. So, this morning I built an antiX image that just has a loop running, and restarts eDEX in fluxbox if not present. not much heavier than just a bare x server.110 MB in my vm I think, and the snapshot is probably around 700MB total. 

idk if you're familiar with antiX, but there isn't actually a large difference between a full-fledged distro and a redistributable respin. 

I'll toss a better one together in a few with doom/evil, ranger, and midnight commander, and maybe switch to an MX base for systemd.

**Comment by technomancer21** [comment]
I'm not familiar with it at all actually. Part of this project was to learn just how to do everything via commandline, but so far all I've managed to do is come up empty at the last step. I'm going to hazard a guess at there being something missing in my Xorg install. it tries to launch but seems to hang somewhere between starting X and launching edex itself. So i'm starting to wonder if I should look more into that. I only ran apt-get install xorg, nothing else. Can you list what you had to install to get it running there john?

**Comment by JohnDanielScanlon** [comment]
Well my solution was a lot easier, as I'm still loading a window manager to not have to mess with .xinit and the fact that it is a node appimage lol. I'm simply running a bash loop that looks for the pid, thus letting it stay booted when it is crash-tastic. I'll upload my snapshot so you can try it out, if you'd like.

I'm posting my shell script here. https://github.com/JohnDanielScanlon/Scripts/blob/main/Bash/AntiDex.sh

Or you can reach me on Discord @onwhAt#5417

**Comment by technomancer21** [comment]
Running a window manager in the background would result in a fair performance drain, would it not?

**Comment by scanlon-dev** [comment]
Not really. Especially not simply a window manager.

It would use maybe 1/100th of what eDEX uses in resources, but probably far, far less. Blackbox/Fluxbox were made for low-end computers in 2001 or before... When 512mb of ram was legendary. So it uses around 4.4mb or ram at startx from what I just googled, and less cpu time than many of your background services. 

Here's my estimate:

Fluxbox 4-10mb
basesystem - 60-90mb?
edex 150-200mb

I was hovering around 280 running two terminals in edex. These are based off of guessing though, so; grain of salt. 

Also I'm trying to not blow up this dev's notifications, so message me on discord or the eDEX shell script I linked.
